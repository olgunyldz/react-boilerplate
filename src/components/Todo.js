import React from 'react';

function Todo({ text }) {
  console.log(text);
  return (
    <div data-testid='todo-item' id='todo-id'>
      {text}
    </div>
  );
}

export default Todo;
