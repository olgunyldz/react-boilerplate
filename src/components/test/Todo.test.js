import { screen, render, cleanup } from '@testing-library/react';
import Todo from '../Todo';

afterEach(() => {
  cleanup();
});

test('should render Todo Component color is green if completed is false', () => {
  const todo = {
    id: 1,
    description: 'React Öğren',
    completed: false,
  };
  render(<Todo todo={todo} />);
  const element = screen.getByTestId('todo-item-1');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent(todo.description);

  expect(element.firstChild).toHaveStyle({ color: 'green' });
});

test('should render Todo Component color is red if completed is true', () => {
  const todo = {
    id: 1,
    description: 'React Öğren',
    completed: true,
  };
  render(<Todo todo={todo} />);
  const element = screen.getByTestId('todo-item-1');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent(todo.description);

  expect(element.firstChild).toHaveStyle({ color: 'red' });
});

test('should render Todo Component as native', () => {
  const todo = {
    id: 1,
    description: 'React Öğren',
    completed: false,
  };
  render(<Todo todo={todo} />);
  const element = document.getElementById('todo-id-1');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent(todo.description);
});
