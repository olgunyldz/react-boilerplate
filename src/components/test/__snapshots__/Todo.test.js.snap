// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should be same 1`] = `
<div
  data-testid="todo-item-1"
  id="todo-id-1"
>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
     
    <strike>
      React Öğren
    </strike>
  </h1>
</div>
`;

exports[`snapshot should be same if completed false 1`] = `
<div
  data-testid="todo-item-2"
  id="todo-id-2"
>
  <h1
    style={
      Object {
        "color": "green",
      }
    }
  >
     
    React Öğren - Advanced
  </h1>
</div>
`;

exports[`snapshot should be same if completed true 1`] = `
<div
  data-testid="todo-item-1"
  id="todo-id-1"
>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
     
    <strike>
      React Öğren
    </strike>
  </h1>
</div>
`;
